### PR TITLE
EIP1-5615 / EIP1-5658 - Happy path for re-issuing an AED

### PR DESCRIPTION
**Pull request by NathanRussellValtech**

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt [hunk 1]
```diff
@@ -41,15 +44,16 @@ class AnonymousElectorDocumentService(
...
-            .sortedByDescending { it.dateCreated }
+        val mostRecentAed = getAnonymousElectorDocumentsSortedByDate(gssCodes, dto.sourceReference)
             .firstOrNull() ?: throw CertificateNotFoundException(eroId, ANONYMOUS_ELECTOR_DOCUMENT, dto.sourceReference)
 
-        TODO("not yet implemented")
+        val templateFilename = pdfTemplateDetailsFactory.getTemplateFilename(mostRecentAed.gssCode)
+        with(reIssueAnonymousElectorDocumentMapper.toNewAnonymousElectorDocument(mostRecentAed, templateFilename)) {
+            val templateDetails = pdfTemplateDetailsFactory.getTemplateDetails(this)
+            val contents = pdfFactory.createPdfContents(templateDetails)
+            anonymousElectorDocumentRepository.save(this)
```
Comment (vishalgupta-valtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt:54]: Is there any possibility to refactor this logic with existing generateAed service function?

Comment (NathanRussellValtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt:54]: Almost certainly, yes 👍 
I'll take a look 👍

File: src/main/kotlin/uk/gov/dluhc/printapi/mapper/aed/AedMappingHelper.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.dluhc.printapi.mapper.aed
+
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.database.entity.AnonymousElectorDocumentStatus
+import java.time.Clock
+import java.time.Instant
+import java.time.LocalDate
+
+@Component
+class AedMappingHelper(private val clock: Clock) {
```
Comment (vishalgupta-valtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/mapper/aed/AedMappingHelper.kt:10]: 👍🏼 nice

File: src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt (hunk 1 above)
Comment (NathanRussellValtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt]: Done 👍

**Review by vishalgupta-valtech — APPROVED** [review]
Lgtm , thanks for refactoring common logic

**Pull request merged**